### ISSUE TITLE
Angular support postcss?

### ISSUE THREAD

**Issue by jon9090** [angular/angular-cli#22849]

**Comment by alan-agius4** [comment]
Angular CLI does use postcss on stylesheets and it comes out of the box which doesn't require any configuration other than configuring the browsers that you'd like to support in the browserslist configuration RC file. 

Regarding CSS-in-JS for components, this is expected and is design choice. One of the reasons for this is to allow view encapsulation since CSS will be emitted with placeholders which will be changed during runtime.

**CSS in JS**
```js
styles: ["a[_ngcontent-%COMP%]{color:#fff}"],
``` 

**Appended css in head**
```html
<style>a[_ngcontent-ubh-c16]{color:#fff}</style>
```

**Comment by jon9090** [comment]
@alan-agius4 I want to use `postcss-custom-media` plugin from https://github.com/postcss/postcss-custom-media
Which gives me custom variables and `postcss` know how to parse that. 

My question is can I add `postcss.config.js` and it's parse every css in my angular application? or angular doesn't support this? if not can it be as feature request?

```
@custom-media --OSdark        (prefers-color-scheme: dark);

@media (--OSdark) {
  html {
    background: var(--gray-9);
    color: var(--gray-1);
  }
}
```

**Comment by alan-agius4** [comment]
> It is useless to create a new postcss.config.js in the root directory. Can you directly configure postcss plugins such as postcss-nesting without using tools such as custom-webpack?

No, Angular CLI itself does not allow configuration of postcss plugins. The fact that Angular is using postcss is an implementation detail.
